packages/hardhat/test/e2e/periphery/controller-helper.ts

```diff
@@ -0,0 +1,489 @@
+      expect(vaultAfter.shortAmount.sub(mintWSqueethAmount).lte(1)).to.be.true
+      expect(vaultAfter.collateralAmount.eq(BigNumber.from(0))).to.be.true
+    })
+
+    describe("open short with already overcollateralized vault, mint, LP oSQTH only, deposit LP NFT and withdraw ETH collateral", async ()=> {
+      before(async () => {
+        const normFactor = await controller.getExpectedNormalizationFactor()
+        const mintWSqueethAmountToLp : BigNumber = ethers.utils.parseUnits('35')
+        const mintRSqueethAmount = mintWSqueethAmountToLp.mul(normFactor).div(one)
+        const ethPrice = await oracle.getTwap(ethUsdcPool.address, weth.address, usdc.address, 420, true)
+        const scaledEthPrice = ethPrice.div(10000)
+        const debtInEth = mintRSqueethAmount.mul(scaledEthPrice).div(one)
+        const collateralAmount = debtInEth.mul(3)
+  
+        await controller.connect(depositor).mintWPowerPerpAmount(0, mintWSqueethAmountToLp, 0, {value: collateralAmount})  
+      })
+
+      it("mint short, LP and use it as collateral", async () => {
+        const vaultId = (await shortSqueeth.nextId()).sub(1);
+        await controller.connect(depositor).updateOperator(vaultId, controllerHelper.address);
+
+        const mintWSqueethAmount = ethers.utils.parseUnits('5')
+        const collateralToMint = BigNumber.from(0)
+        const collateralToLp = BigNumber.from(0)
+        const flashloanWMintDepositNftParams = {
+          vaultId: vaultId,
+          wPowerPerpAmount: mintWSqueethAmount.toString(),
+          collateralToDeposit: collateralToMint.toString(),
+          collateralToFlashloan: BigNumber.from(0),
+          collateralToLp: collateralToLp.toString(),
+          collateralToWithdraw: 0,
+          lpAmount0Min: 0,
+          lpAmount1Min: 0,
+          lpLowerTick: -887220,
+          lpUpperTick: 0
+        }
+
+        const vaultBefore = await controller.vaults(vaultId)
+
+        await controllerHelper.connect(depositor).flashloanWMintDepositNft(flashloanWMintDepositNftParams)
+
+        const vaultAfter = await controller.vaults(vaultId)
+        const tokenIndexAfter = await (positionManager as INonfungiblePositionManager).totalSupply();
+        const tokenId = await (positionManager as INonfungiblePositionManager).tokenByIndex(tokenIndexAfter.sub(1));
+        const position = await (positionManager as INonfungiblePositionManager).positions(tokenId)
+
+        expect(BigNumber.from(vaultAfter.NftCollateralId).eq(tokenId)).to.be.true;
+        expect(position.tickLower === -887220).to.be.true
+        expect(position.tickUpper === 0).to.be.true
+        expect(vaultAfter.shortAmount.sub(vaultBefore.shortAmount).eq(mintWSqueethAmount)).to.be.true
+        expect(vaultAfter.collateralAmount.eq(vaultBefore.collateralAmount)).to.be.true
+      })
+    })
+
+    it("open short in new vault, mint, LP oSQTH only, deposit LP NFT and withdraw ETH collateral", async () => {
```

Comment:
no collateral is withdrawn here, right?

Reply:
this case is that user deposits at 75% CR ratio before depositing the NFT (and only flashloands 1/2 of the required CR before LPing and depositing), right?

Reply:
maybe add mint with >0 ETH collateral

Reply:
Yes